src/main/java/org/hydev/mcpm/client/installer/InstallInteractor.java

```diff
@@ -73,36 +82,59 @@ public boolean installPlugin(InstallInput installInput,
         var idPluginModel = pluginModel.id();
         var pluginVersion = pluginModel.getLatestPluginVersion().orElse(null);
         if (pluginVersion == null) {
-            installPresenter.displayResult(new InstallResult(Type.NO_VERSION_AVAILABLE), installInput.name());
-            return false;
+            return List.of(new InstallResult(Type.NO_VERSION_AVAILABLE, pluginName));
         }
 
-        // Name of the latest version plugin
-        String pluginName = pluginVersion.meta().name();
-
-        if (databaseManager.checkPluginInstalledByName(pluginName)) {
-            InstallResult installResult = new InstallResult(Type.PLUGIN_EXISTS);
-            installPresenter.displayResult(installResult, installInput.name());
+        var results = new ArrayList<InstallResult>();
+        File pluginFile = superPluginTracker.getPluginFile(pluginName);
+        if (pluginFile != null) {
+            results.add(new InstallResult(Type.PLUGIN_EXISTS, pluginName));
         } else {
             // 3. Download it
-            spigotPluginDownloader.download(idPluginModel, pluginVersion.id(),
-                    "plugins/" + pluginVersion.meta().name() + ".jar");
-            databaseManager.addManualInstalled(pluginVersion, installInput.isManuallyInstalled());
-
-            // 4. Load the plugin
-            loadPlugin(installInput, installPresenter);
+            spigotPluginDownloader.download(pluginName, idPluginModel, pluginVersion.id());
+            // 4. Add the installed plugin to the json file
```

Comment:
Is this commented out because something is held up with the SuperPluginTracker? but it got merged